test/parallel/test-child-process-fork-timeout-kill-signal.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+const { mustCall } = require('../common');
+const { strictEqual, ok, throws } = require('assert');
+const fixtures = require('../common/fixtures');
+const { fork } = require('child_process');
+const { getEventListeners } = require('events');
+
+{
+  // Verify default signal + closes after at least 4 ms.
+  let closed = false;
+  setTimeout(() => ok(!closed), 4);
```

Comment:
This could end up flaky on some systems. Relying on precise timing on asserts is... unreliable.

Reply:
Thanks, I guess that just checking that it gets killed correctly will be enough.
Once #37325 lands, I'll do some cleanup on this PR and make it ready.